addons/purchase_stock/tests/test_average_price.py

```diff
@@ -254,3 +254,53 @@ def test_bill_before_reciept(self):
         picking.button_validate()
 
         self.assertEqual(avco_product.avg_cost, 300)
+
+    def test_can_reset_vendor_bill_avco_product_high_quantity_order(self):
```

Comment:
I think what does really matter here is the broken valuation. Not being able to reset the invoice is just a consequence of that. So, I would rather focus on that part of the use case and ignore the reset part (as you did in the commit redaction, actually)
-> Rename the test and improve the asserts